app/vlinsert/main.go

```diff
@@ -29,15 +36,27 @@ func Stop() {
 func RequestHandler(w http.ResponseWriter, r *http.Request) bool {
 	path := r.URL.Path
 
-	if path == "/internal/insert" {
+	switch {
+	case strings.HasPrefix(path, "/internal/insert/") || path == "/internal/insert":
+		if *disableInternal {
```

Comment:
The `-insert.disable` flag must also disable requests to `/internal/insert`, even if `-internalinsert.disable` isn't set, since this is what most users will expect.